feat: Add uniswapv3-base

Eric-Warehime opened the pull request:
Add uniswapv3-base and some basic configuration around it.

Eric-Warehime committed:
Add uniswapv3-base

aljo242 reviewed (approved):
lgtm @Eric-Warehime can we run a test deployment?

Commit:
Merge branch 'main' into eric/uniswap-base

g3ndo9kar9 commented on providers/apis/defi/uniswapv3/utils.go:
```diff
@@ -96,6 +97,44 @@ var (
+
+	// DefaultBaseMarketConfig is the default market configuration for Uniswap V3. Specifically
+	// this is for Base mainnet.
+	DefaultBaseMarketConfig = types.CurrencyPairsToProviderTickers{
```
Can we remove the default market config? My plan is to deprecate all of these + use our static marketmaps for testing

Commit:
Merge branch 'main' into eric/uniswap-base

Zygimantass committed:
fix(ci): debug private key issue (#465)

* fix: debug

* fix: change tag ordering

* fix: remove pr trigger

davidterpay committed:
chore: Deprecate URL in our configurations (#461)

* init

* nit

* kek

* nit

* ws test

* nits

---------

Co-authored-by: Alex Johnson <alex@skip.money>

davidterpay committed:
chore: Add Core + Raydium Markets into Exportable Go structs (#466)

* init

* rm old market.json

* test case

* rm bitstamp and bitfinex

* nit

g3ndo9kar9 committed:
feat: remove all references to oracle-config-path [BLO-1313] (#463)

* remove all references to oracle-config-path

* remove all oracle.json files

* add entrypoints to docker images

---------

Co-authored-by: Alex Johnson <alex@skip.money>
Co-authored-by: David Terpay <35130517+davidterpay@users.noreply.github.com>

g3ndo9kar9 committed:
feat: remove slinky-config cmd [BLO-1320] (#468)

* remove slinky-config cmd

* update e2e dockerfile

---------

Co-authored-by: Alex Johnson <alex@skip.money>

davidterpay committed:
fix: dYdX Provider to use correct query params (#471)

* fix

* add http prefix

g3ndo9kar9 committed:
feat: Make `OracleConfig.Providers` a map instead of an array [BLO-1318] (#467)

* remove all references to oracle-config-path

* move providers to be a map in slinky config

* remove Production field

---------

Co-authored-by: Alex Johnson <alex@skip.money>

dependabot[bot] committed:
chore(deps): bump github.com/spf13/viper from 1.18.2 to 1.19.0 (#474)

Bumps [github.com/spf13/viper](https://github.com/spf13/viper) from 1.18.2 to 1.19.0.
- [Release notes](https://github.com/spf13/viper/releases)
- [Commits](https://github.com/spf13/viper/compare/v1.18.2...v1.19.0)

---
updated-dependencies:
- dependency-name: github.com/spf13/viper
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

g3ndo9kar9 committed:
remove oracle.json mount (#472)

davidterpay committed:
chore: Removing market json files in repo (#470)

* init

* rm old market.json

* test case

* rm bitstamp and bitfinex

* nit

* rm dirs

* more changes

* adding pre-gen script

* docker compose changes

* nits

* nit provider merges

* phony

* conditional set of env vars

davidterpay committed:
fix: No Proxy on Marketmap Provider (#477)

* init

* aisdf

* asdf

* no proxy on marketmap provider

* nit

* nit

Eric-Warehime committed:
Update base markets

Eric-Warehime committed:
Update values

Eric-Warehime committed:
Merge remote-tracking branch 'origin/main' into eric/uniswap-base

davidterpay committed:
Merge branch 'main' into eric/uniswap-base

davidterpay committed:
update genesis script

davidterpay committed:
make update

Eric-Warehime committed:
bump

davidterpay commented:
Can you update the dYdX provider to accept feeds from base? Also have we decided on the format of what the ticker will look like in their gov proposals?

Eric-Warehime committed:
Add dydx parse logic for uniswapv3 base

Eric-Warehime committed:
Invert brett/eth

Eric-Warehime committed:
Merge branch 'main' into eric/uniswap-base

davidterpay commented on Makefile:
```diff
@@ -17,8 +17,9 @@ APP_TOML ?= $(HOMEDIR)/config/app.toml
 COVER_FILE ?= cover.out
 BENCHMARK_ITERS ?= 10
-USE_CORE_MARKETS ?= true
+USE_CORE_MARKETS ?= false
 USE_RAYDIUM_MARKETS ?= false
+USE_UNISWAPV3_BASE_MARKETS ?= true
```
nit: Would either default all of these to true or false.

g3ndo9kar9 commented on cmd/constants/markets.go:
```diff
@@ -5161,6 +5161,37 @@ var (
+	UniswapV3BaseMarketMap mmtypes.MarketMap
+
+	// UniswapV3BaseMarketMapJSON is the JSON representation of UniswapV3BaseMarketMap.
+	UniswapV3BaseMarketMapJSON = `
```
nit: is this a sufficient # of markets from base to give us confidence in the provider's operation?

Eric-Warehime replied:
Added some more :D

g3ndo9kar9 commented on providers/apis/defi/uniswapv3/utils.go:
```diff
@@ -84,15 +88,30 @@ func (pc PoolConfig) MustToJSON() string {
+var (
+	// DefaultETHAPIConfig is the default configuration for the Uniswap API. Specifically this is for
+	// Ethereum mainnet.
+	DefaultETHAPIConfig = config.APIConfig{
```
nit: could change provider default name from `BaseName` to smth that doesn't contain `Base`

g3ndo9kar9 reviewed (approved):
lgtm

Eric-Warehime committed:
Add more interesting test markets
